### ISSUE TITLE
General issues with converting from conda to pixi

### ISSUE THREAD

**Issue by vigneshmanick** [prefix-dev/pixi#1058]
### Problem description

Hello Pixi devs,

I have listed some issues that i have faced when testing moving to pixi from conda.

1. The install script should have an option to explicity specify the pixi install location. Defaulting to `$HOME` is fine but in corporate environments the `home` folder is very limited in space/locked down/automatic cleanup etc and generally the dev tools are installed on local drives (e.g /scratch).  I solved it by setting the `PIXI_HOME` env variable, but this is information is not readily apparent in the documentation.
2.  Also related to the above point, the `PIXI_HOME` could be automatically set by pixi when an install loation is specified
3. The auto complete  magic `echo 'eval "$(pixi completion --shell bash)"' >> ~/.bashrc` is appended to the end of the `.bashrc ` file. This is fine for cases where the file doesn't contain conda activation block but for cases where a conda activation block is present, the modifications should be above this section since conda recommends it's data should be at the end of the file. To be specific, it would be good if pixi related changes are made above the following lines to reduce friction in moving from conda

```
# >>> conda initialize >>>
# !! Contents within this block are managed by 'conda init' !!
```
5. After installation, the pixi activation is added to `bash_profile` and not `.bashrc`, this will result in non login shells not having pixi avaialble. Conda adds to `.bashrc` and not to `bash_profile`
6. After installation i tried to use the recommended method to import a conda env with the command `pixi init --import environment.yml` and i got the following error

```
  × The current system has a mismatching virtual package. The project requires '__linux' to be at least version '5.10' but the system has version
  │ '3.10.0'
```

I searched the issues and found that the following needs to be set which is not readily apparent in the documentation. If i have made any mistakes, please feel free to correct me.

```
[system-requirements]
linux="3.10.0'"
```
 If you ask why the version is so low, we use RHEL-7.9 and this is generally the standard in HPC, Uni / Research institutes. And soon it will be upgraded to RHEL-8.9 which has the version ` RHEL 8.9 | 4.18.0-513.5.1.el8_9` .  Conda packages are built with the conda compilers which themselves have very low glibc requirements right? so the packages installed by pixi via conda should have no issues with these `uname` versions. I t would be good if the above error is removed and the required block directly added to the `pixi.toml`

7. The most important of all, to connect to internet the machines use proxy and also a custom ca-certificate. This is configured in the `.condarc` file via the proxy setting and also via `conda config --set ssl_verify <pathToYourFile>.crt` . There is no possiblity for the users to update these settings globally. It would be good if pixi also would use the same/similar configuration for connection. As of now, the install works only when i am not in the office network. I have also detailed the erros for this in https://github.com/prefix-dev/pixi/issues/1035.

Thanks for your awesome tool. I am testing this and working with it successfully in for `home office` setting and hope to be able to use it also in `office` setting.

**Comment by ruben-arts** [comment]
Thank you @vigneshmanick, this is really useful information, I'll take a more in depth look later and probably move some items into issues or PR's!

**Comment by vigneshmanick** [comment]
@ruben-arts i have some further points regarding the issue. Would you prefer that i update the issue description here or create another ?

**Comment by tdejager** [comment]
Please update it here, I don't think we've gotten around creating individual issues out of them yet :) Thanks for all the testing!

**Comment by vigneshmanick** [comment]
Have updated the description and might add more points as and when i face them.

**Comment by j-bac** [comment]
Talking about requirements.txt, it would be useful to be able to pixi add requirements.txt or pixi init --import requirements.txt

**Comment by kxmh42** [comment]
I have a problem related to this issue: PATH is modified by the Pixi install script only in `.bashrc`, and in most systems `.bashrc` is configured not to work in non-interactive shells, which means I can't execute remote commands such as `ssh myhost pixi global add blabla`, as they are executed in non-interactive shells.

The Pixi install script should modify PATH not only in non-login startup files as it does now (`.bashrc`, `.zshrc`, etc.), but also in login startup files (`.profile`, `.bash_profile`, `.zlogin`, etc.). I'd suggest following the logic used by [userpath](https://github.com/ofek/userpath), which is a standard tool for doing such things.

**Comment by kxmh42** [comment]
I investigated it a bit more, and it turns out that it is annoyingly hard to properly add something to PATH so that it works in all typical circumstances. Basically we want the PATH to be updated in both interactive and non-interactive shells, we should add the appropriate commands:
- in `bash`: to the *beginning* of `.bashrc` (before the check if shell is interactive, which is usually at the top of the file)
- in `tcsh`: to the *beginning* of `.tcshrc` (before the check if shell is interactive, which is usually at the top of the file)
- in `zsh`: to `.zshenv` (which is always executed, unlike `.zshrc`, which is only executed by interactive shells)

In other shells, such as `fish` and `xonsh`, the rc files are always executed, so appropriate commands can probably be added to the end of the rc file.

I can submit a PR if needed.

**Comment by vigneshmanick** [comment]
for bash you could use `$BASH_ENV` to set values for non-interactive shells. I don't think it's a good idea to put any values before the check for interactive/non interactive is done. Generally all values in bashrc are set after this check and pixi is no exception here.

**Comment by kxmh42** [comment]
I agree, however it would still be better for the Pixi installer to update `.zshenv` instead of `.zshrc`.

**Comment by baszalmstra** [comment]
Im going to close this as stale. Feel free to open new issues for anything that is still blocking you!
